### PR TITLE
Support AutoCloseable in closeAll

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

**Review by losipiuk — APPROVED** [review]

File: testing/src/main/java/io/airlift/testing/Closeables.java [hunk 1]
```diff
@@ -35,12 +36,25 @@ public static void closeQuietly(Closeable... closeables)
...
+            closeAll((AutoCloseable[]) closeables);
+        }
+        catch (Exception e) {
+            propagateIfPossible(e, IOException.class);
+            // Not possible
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void closeAll(AutoCloseable... closeables)
```
Comment (electrum) [inline testing/src/main/java/io/airlift/testing/Closeables.java:50]: The usages of `IOException` need to be changed

Comment (findepi) [reply testing/src/main/java/io/airlift/testing/Closeables.java:50]: This is why i retained `closeAll(Closeable...)`

Comment (findepi) [reply testing/src/main/java/io/airlift/testing/Closeables.java:50]: BTW I'd rather remove `closeAll(Closeable...)` if we can afford breakage.
(there is no reasonable gradual deprecation path I can think of)

Comment (martint) [reply testing/src/main/java/io/airlift/testing/Closeables.java:50]: In the code below, we shouldn't wrap the exception with an IOException for this method:

```java
 catch (Throwable e) {
    if (rootCause == null) {
        rootCause = new IOException(e);
    }
```

Comment (findepi) [reply testing/src/main/java/io/airlift/testing/Closeables.java:50]: good catch

**Comment by findepi** [comment]
AC

**Review by martint — APPROVED** [review]

**Pull request merged**